src/components/Filters/index.tsx

```diff
@@ -66,482 +30,293 @@ const FilterWrapper = styled(Box)`
+		const { schema } = filtersState;
+		const filter = SchemaSieve.createFilter(schema, edit);
 
-		let filters: JSONSchema6[];
+		let currentFilters: JSONSchema6[];
 
 		if (!!$id) {
-			const matchIndex = findIndex(currentFilters, { $id });
-			currentFilters.splice(matchIndex, 1, filter);
-			filters = currentFilters.slice();
+			const matchIndex = findIndex(filtersState.filters, { $id });
+			filtersState.filters.splice(matchIndex, 1, filter);
+			currentFilters = filtersState.filters.slice();
 		} else {
-			filters = currentFilters.concat(filter);
+			currentFilters = filtersState.filters.concat(filter);
 		}
 
-		this.setState(
-			{
-				filters,
-				edit: [this.getCleanEditModel()],
-				showModal: false,
-				editingFilter: null,
-			},
-			() => this.emitFilterUpdate(),
-		);
-	}
+		setFiltersState(prevState => ({
+			...prevState,
+			filters: currentFilters,
+			edit: [SchemaSieve.getCleanEditModel(filtersState.schema)],
+			showModal: false,
+			editingFilter: null,
+		}));
+	};
 
-	public editFilter(filter: JSONSchema6) {
-		const { schema } = this.state;
+	const editFilter = (filter: JSONSchema6) => {
+		const { schema } = filtersState;
 
 		const signatures = SchemaSieve.decodeFilter(schema, filter);
 
-		this.setState({
+		setFiltersState(prevState => ({
+			...prevState,
 			edit: signatures as EditModel[],
 			editingFilter: filter.$id!,
 			showModal: true,
-		});
-	}
-
-	public removeFilter({ $id, title }: JSONSchema6) {
-		this.setState(
-			prevState => {
-				const newState = {
-					...prevState,
-					filters: reject(prevState.filters, { $id }),
-				};
-
-				if (title === SchemaSieve.FULL_TEXT_SLUG) {
-					newState.searchString = '';
-				}
-
-				return newState;
-			},
-			() => this.emitFilterUpdate(),
-		);
-	}
+		}));
+	};
 
-	public setFilters(filters: JSONSchema6[]) {
-		this.setState({ filters }, () => this.emitFilterUpdate());
-	}
+	const removeFilter = ({ $id, title }: JSONSchema6) => {
+		const newState = {
+			...filtersState,
+			filters: reject(filtersState.filters, { $id }),
+		};
 
-	public clearAllFilters = () => {
-		this.setFilters([]);
-		this.setState({
-			searchString: '',
-		});
+		if (title === SchemaSieve.FULL_TEXT_SLUG) {
+			newState.searchString = '';
+		}
+		setFiltersState(newState);
 	};
 
-	public addCompound() {
-		this.setState(prevState => ({
-			edit: prevState.edit.concat(this.getCleanEditModel()),
+	const clearAllFilters = () => {
+		setFiltersState(prevState => ({
+			...prevState,
+			filters: [],
+			searchString: '',
 		}));
-	}
-
-	public removeCompound(index: number) {
-		const edit = this.state.edit.slice();
-		edit.splice(index, 1);
-		this.setState({
-			edit,
-		});
-	}
+	};
 
-	public saveView(name: string, scope: string | null) {
+	const saveView = (name: string, scope: string | null) => {
 		const view: FiltersView = {
 			id: randomString(),
 			name,
 			scope,
-			filters: cloneDeep(this.state.filters),
+			filters: cloneDeep(filtersState.filters),
```

Comment:
Why do we clone the filters here? Ideally, we should never do deep clones, and deep comparisons, and just recreate objects as deep as necessary when they change.